hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -1020,8 +1020,8 @@ namespace hazelcast {
                 std::string operation_name_;
 
                 std::vector<std::vector<byte>> data_buffer_;
-                size_t buffer_index_;
-                size_t offset_;
+                size_t buffer_index_ {0};
+                size_t offset_ {0};
```

Comment:
They are incremented after this during client message write methods (see method `peek`). C++ does not zero initialize by default (probably due to performance), and it can be any arbitrary value actually depending on the memory region.

Reply:
But is this related to this PR? So it was a bug I guess.